fix/update-local-setup-doc (#98)
RandellDawson/github-bots: docs/README.md

## Local Setup
- Follow the steps below to get this running on your local machine

### 1. Copy .env
- Copy the `sample.env` file into `.env`. The command below will do that in the terminal if your CWD(current working directory) is the `contribute` folder.
```bash
cp probot/sample.env probot/.env
```
- If you do not want to populate the database with the freeCodeCamp PR's you can [skip to step 6](#6-start-the-program)

### 2. Update .env
- Put your GitHub username in the `GITHUB_USERNAME` field of the `.env` file

### 3. Obtain GitHub Personal access token
- While on Github, click your profile icon > `Settings`
- Then click `Developer settings` > `Personal access tokens`
- Click `Generate new token`
- On the next page, give the token a name so you know what it’s for
- Scroll down to the bottom and click `Generate token`
- Copy the token string and paste it into the `GITHUB_ACCESS_TOKEN` field in the `.env` file. Note that you will not be able to see this token string on GitHub again

### 4. Create your own GitHub app
- While on GitHub, click your profile icon > `Settings`
- Then click `developer settings` > `New GitHub app`

> - Fill in the `name` field with a name of your choice

Reviewer: ```suggestion
- Fill in the `name` field with a name of your choice
- Fill in the `homepage` field with `https://contribute.freecodecamp.org`
```
Does it matter what we put in there?  I wasn't sure, so I went with the contribute url.

Author: @RandellDawson suggested to me to use `https://localhost` - that's what I put in there. I didn't realize it can't be blank so we should probably suggest to put something in there.

Reviewer: Seems it doesn't matter much, but I did a bit of googling, found probot's [guide](https://probot.github.io/docs/deployment/#create-the-github-app) and they say:
> Homepage URL: the URL to the GitHub repository for your app

For the sake of putting something, how about we say that?

Author: so something like.. 

- Fill in the `homepage` field with the URL to the GitHub repository for your app. e.g.`https://github.com/username/contribute`

Reviewer: Yeah, that seems reasonable.

## Local Setup
- Follow the steps below to get this running on your local machine

### 1. Copy .env
- Copy the `sample.env` file into `.env`. The command below will do that in the terminal if your CWD(current working directory) is the `contribute` folder.
```bash
cp probot/sample.env probot/.env
```
- If you do not want to populate the database with the freeCodeCamp PR's you can [skip to step 6](#6-start-the-program)

### 2. Update .env

> - Put your GitHub username in the `username` field of the `.env` file

Reviewer 2: ```suggestion
- Put your GitHub username in the `GITHUB_USERNAME` field of the `.env` file
```

### 3. Obtain GitHub Personal access token
- While on Github, click your profile icon > `Settings`
- Then click `Developer settings` > `Personal access tokens`
- Click `Generate new token`
- On the next page, give the token a name so you know what it’s for
- Scroll down to the bottom and click `Generate token`
- Copy the token string and paste it into the `GITHUB_ACCESS_TOKEN` field in the `.env` file. Note that you will not be able to see this token string on GitHub again

### 4. Create your own GitHub app
- While on GitHub, click your profile icon > `Settings`
- Then click `developer settings` > `New GitHub app`
- Fill in the `name` field with a name of your choice

> - Fill in the `Homepage URL` field with `https://localhost`

Reviewer 2: This line can actually be left out
```suggestion
```

- In a new tab, go to https://smee.io/

> - Click `Start a new channel` and copy the URL they give you. You can ignore the rest of the instructions on this page

Reviewer 2: ```suggestion
- Click `Start a new channel` and copy the URL they give you. You can ignore the rest of the instructions on the `smee.io` page
```

- Scroll to the bottom of your GitHub app page and click `Create GitHub App`
- On the next page, copy the `App ID` and paste it into the `APP_ID` field of the `.env` file
- Scroll to the bottom of this page and click `Generate Private Key`
- A popup menu will allow you to download the key file. Download it and put it in the `probot` folder

### 5. Run mongoDB
- Make sure a mongoDB instance is running by running the command below in the terminal.
```bash
mongod —dbpath=./database_folder
```

### 6. Start the program
- In a new terminal window or tab, run these three commands to start the program. Wait for one command to finish running before starting the next.
```bash
npm install
npm run build
npm run develop
```
- If you skipped steps 2-5, you can ignore the rest

### 7. Update the Database
- Run the command below to populate your local database with PR’s from the freeCodeCamp repo. Note that you must have the program running and mongoDB running.
```bash
node probot/server/tools/update-db.js
```
- This will take a while. If it stops running partway through, it's probably a timeout error. Run the command again and it should finish

> - After this is complete you will need to re-run `npm run develop` to update the program with the new info from the DB

Reviewer 2: This should not be necessary as it just seeds the database.
```suggestion
```
